agent/internal/request.go

```diff
@@ -165,13 +166,26 @@ func (ctx *HTTPRequestRecord) NewUserSignup(id map[string]string) {
 }
 
 func (ctx *HTTPRequestRecord) Close() {
-	ctx.agent.addTrackEvent(newHTTPRequestRecord(ctx))
+	if !ctx.shouldSend {
+		return
+	}
+
+	ctx.agent.addRecord(newHTTPRequestRecord(ctx))
+}
+
+func (ctx *HTTPRequestRecord) addSilentEvent(event *HTTPRequestEvent) {
+	ctx.addEvent_(event, true)
 }
 
 func (ctx *HTTPRequestRecord) addEvent(event *HTTPRequestEvent) {
+	ctx.addEvent_(event, false)
+}
+
+func (ctx *HTTPRequestRecord) addEvent_(event *HTTPRequestEvent, silent bool) {
 	ctx.eventsLock.Lock()
 	defer ctx.eventsLock.Unlock()
 	ctx.events = append(ctx.events, event)
+	ctx.shouldSend = !silent
```

Comment:
Where is `silent` defined?

Reply:
It relies on the zero value of a boolean, which is `false`.